packages/tradershub/src/hooks/useRegulationSwitcher.ts

```diff
@@ -0,0 +1,58 @@
+import { useEffect } from 'react';
+import { useActiveTradingAccount, useAuthorize, useTradingAccountsList } from '@deriv/api';
+import { useUIContext } from '../components';
+import { Regulation } from '../constants/constants';
+import useRegulationFlags from './useRegulationFlags';
+
+/**
+ * @description This hook contains the logic that is used to switch between EU and non-EU accounts
+ * @returns  {buttons: {label: string}[], handleButtonClick: (label: string) => void}
+ * @example
+ * const { buttons, handleButtonClick } = useRegulationSwitcher();
+ */
+export const useRegulationSwitcher = () => {
+    const { switchAccount } = useAuthorize();
+    const { data: tradingAccountsList } = useTradingAccountsList();
+    const { getUIState, setUIState } = useUIContext();
+
+    const regulation = getUIState('regulation');
+    const accountType = getUIState('accountType');
+    const { isEU, isHighRisk } = useRegulationFlags(regulation, accountType);
+
+    const realCRAccount = tradingAccountsList?.find(account => account.loginid.startsWith('CR'))?.loginid ?? '';
+    const realMFAccount = tradingAccountsList?.find(account => account.loginid.startsWith('MF'))?.loginid ?? '';
+
+    const { data: activeTrading } = useActiveTradingAccount();
+
+    const buttons = [{ label: Regulation.NonEU }, { label: Regulation.EU }];
+
+    const handleButtonClick = (label: string) => {
+        if (label === Regulation.NonEU) {
+            setUIState('regulation', Regulation.NonEU);
```

Comment:
@thisyahlen-deriv If the user click Non-EU button a couple of times it will be rerendered everytime here. You have to check current regulation before changing state.